test/server_tests.dart

```diff
@@ -75,6 +75,38 @@ void main() {
       expect(resp.mockContent, equals("response"));
     });
 
+    test("compound group path matching", () async {
+      var req = new MockRequest("/mixed/path/subpath");
+      var req2 = new MockRequest("/mixed/path/anotherpath");
+      var req3 = new MockRequest("/mixed/paths");
+      var req4 = new MockRequest("/mixed");
+      var req5 = new MockRequest("/mixed.json");
+      var req6 = new MockRequest("/mixed", method: POST);
+      var req7 = new MockRequest("/mixed/change_status_code");
+      var req8 = new MockRequest("/mixed/info");
+      var req9 = new MockRequest("/mixed/version");
+
+      var resp = await dispatch(req);
+      expect(resp.mockContent, equals("interceptor sub_route"));
```

Comment:
I think you don't need to use `equals` in those expectations -- the default behavior is `equals`.